storage-provider/server/src/main.rs

```diff
@@ -165,6 +173,25 @@ pub enum ServerError {
 
     #[error(transparent)]
     Json(#[from] serde_json::Error),
+
+    #[error(transparent)]
+    P2P(#[from] P2PError),
+}
+
+/// Takes an expression that returns a nested result and
+/// inspected the error while logging it.
```

Comment:
```suggestion
/// Takes an expression that returns a `Result<Result<T, E2>, E1>`.
/// It tries to inspect and log the first error (`E1`), otherwise,
/// it inspects the result and tries to inspect the nested error (`E2`).
///
/// This macro is *roughly* equivalent to calling:
/// ```text
/// res // : Result<Result<T, E2>, E1>
///     .inspect_err(|e| error!(%e))
///     .inspect(|r| r.inspect_err(|e| error!(%e))
/// ```
```